libs/gson-support/src/main/java/org/codice/gsonsupport/GsonTypes.java

```diff
@@ -0,0 +1,646 @@
+/**
```

Comment:
📓 This class has been brought over from Gson's [`$Gson$Types`](https://github.com/google/gson/blob/master/gson/src/main/java/com/google/gson/internal/$Gson$Types.java) with a small change to inline the calls to the `$Gson$Preconditions` class.

Reply:
That should be commented somewhere in this file instead of only on the PR.